pkg/controller/pods.go

```diff
@@ -0,0 +1,22 @@
+package controller
```

Comment:
`pkg/controller` is rather ambiguous, given most of the code in the repo pertains to the controller.

Maybe move this to `pkg/k8s`, and implement utility functions related to the controller in `pkg/k8s/osm_controller.go`.